code/modules/power/generator.dm

```diff
@@ -135,6 +164,13 @@
 		lastgenlev = genlev
 		update_icon()
 	add_avail(effective_gen)
+	if(reagents.has_reagent(/datum/reagent/lube/oil))
+		reagents.remove_any(0.01)
```

Comment:
Subtle bug - you're not guaranteed to actually use up the _oil_ for lubrication if any other reagent winds up in machine.

Reply:
There is a check on proc that handles adding the oil.